sdk/messaging/azservicebus/receiver_test.go

```diff
@@ -85,6 +85,43 @@ func TestReceiverForceTimeoutWithTooFewMessages(t *testing.T) {
 	require.NoError(t, receiver.CompleteMessage(context.Background(), messages[0]))
 }
 
+func TestReceiverDrainHasError(t *testing.T) {
+	serviceBusClient, cleanup, queueName := setupLiveTest(t, nil)
+	defer cleanup()
+
+	sender, err := serviceBusClient.NewSender(queueName, nil)
+	require.NoError(t, err)
+	defer sender.Close(context.Background())
+
+	err = sender.SendMessage(context.Background(), &Message{
+		Body: []byte("hello"),
+	})
+	require.NoError(t, err)
+
+	receiver, err := serviceBusClient.NewReceiverForQueue(queueName, nil)
+	require.NoError(t, err)
+
+	// close the link _just_ before the drain credit to simulate connection loss.
```

Comment:
The link recovery logic prevents us from seeing this connection hiccup right?

Reply:
In this it's not the link recovery - it's the fact that we received some messages (this falls into the we just represent "partial failure as success" bucket).

I have another changeset to try to do connection recovery but haven't finished that yet.

Reply:
Okay, fixed this a slightly different way.

Basically, if the drain fails the link is detached (or dead) so there's nothing left to drain here. So instead, I'll just return. The next call will properly recover the link/connection if need be.